ewmh.c

```diff
@@ -216,6 +216,7 @@ ewmh_init(void)
     luaA_class_connect_signal(L, &client_class, "property::fullscreen" , ewmh_client_update_hints);
     luaA_class_connect_signal(L, &client_class, "property::maximized_horizontal" , ewmh_client_update_hints);
     luaA_class_connect_signal(L, &client_class, "property::maximized_vertical" , ewmh_client_update_hints);
+    luaA_class_connect_signal(L, &client_class, "property::maximized" , ewmh_client_update_hints);
```

Comment:
So... a client can be maximized vertically, maximized horizontally and at the same time not be maximized? Put differently, these are now three completely independent booleans? And all code that wants to check for `maximized_vertical` should now check `maximized_vertical or maximized`, because the C code no longer does so for you? That seems rather odd and I don't like it at all.

How about deprecating the horizontal/vertical maximization instead and only have a single "maximized" state instead?